core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -90,6 +90,23 @@ static boolean hasIds(Schema schema) {
     return AvroCustomOrderSchemaVisitor.visit(schema, new HasIds());
   }
 
+  /**
+   * @param schema an Avro Schema
+   * @return true/false based on whether any of the nodes in the provided schema is missing an
+   * ID property recognizable by Iceberg core API. To have an ID recognizable by Iceberg core API:
+   * <ul>
+   *   <li>a field node under struct (record) schema should have {@link FIELD_ID_PROP} property
+   *   <li>an element node under list (array) schema should have {@link ELEMENT_ID_PROP} property
+   *   <li>a pair of key and value node under map schema should have {@link KEY_ID_PROP} and
+   *   {@link VALUE_ID_PROP} respectively
+   *   <li>a primitive node is not assigned any ID related properties
+   * </ul>
```

Comment:
This long content is better suited for the main description, rather than in the return. `@return` should be a short summary, like "@return true if the schema has at least one field ID property, false otherwise"

The details about what "field ID property" could be should be above.

Reply:
Moved out of return section and before `param` or `return` in a general description.